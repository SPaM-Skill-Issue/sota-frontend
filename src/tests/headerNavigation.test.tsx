import HeaderNavigation from '../components/headerNavigation';
import { render } from '@testing-library/react';
import { MemoryRouter } from "react-router-dom";
import "@testing-library/jest-dom";

const pagePath = [
    {
        label: 'Home',
        path: '/',
    },
    {
        label: 'Sports',
        path: '/sports',
    },
    {
        label: 'Medal',
        path: '/medal',
    },
    {
        label: 'Audience',
        path: '/audience',
    },
];

describe('HeaderNavigation Component', () => {
    it('should renders correctly', () => {
        const wrapper = render(
            <MemoryRouter>
                <HeaderNavigation />
            </MemoryRouter>
        );
        expect(wrapper).toMatchSnapshot();
    });

    it('should renders with correct active key for every page', () => {
        pagePath.forEach((page) => {
            const { container } = render(
                <MemoryRouter initialEntries={[page.path]}>
                    <HeaderNavigation />
                </MemoryRouter>
            );
            const activeMenuItem = container.querySelector('.ant-menu-item-selected');
            expect(activeMenuItem).toBeInTheDocument();
            expect(activeMenuItem).toHaveTextContent(page.label);
        });
    });
});