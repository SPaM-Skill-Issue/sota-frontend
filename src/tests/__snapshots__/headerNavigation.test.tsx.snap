// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderNavigation Component should renders correctly 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light bg-eeric-black-light w-fit font-primary text-xl css-dev-only-do-not-override-1adbn6x"
        data-menu-list="true"
        role="menu"
        tabindex="0"
      >
        <li
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-selected ant-menu-item-only-child text-white"
          data-menu-id="rc-menu-uuid-test-home"
          role="menuitem"
          style="opacity: 1; order: 0;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <a
              href="/"
            >
              Home
            </a>
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child text-white"
          role="menuitem"
          style="opacity: 1; order: 1;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <a
              href="/sports"
            >
              Sports
            </a>
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child text-white"
          role="menuitem"
          style="opacity: 1; order: 2;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <a
              href="/medal"
            >
              Medal
            </a>
          </span>
        </li>
        <li
          class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child text-white"
          role="menuitem"
          style="opacity: 1; order: 3;"
          tabindex="-1"
        >
          <span
            class="ant-menu-title-content"
          >
            <a
              href="/audience"
            >
              Audience
            </a>
          </span>
        </li>
        <li
          aria-hidden="true"
          class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
          role="none"
          style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
        >
          <div
            aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
            aria-expanded="false"
            aria-haspopup="true"
            class="ant-menu-submenu-title"
            data-menu-id="rc-menu-uuid-test-rc-menu-more"
            role="menuitem"
            tabindex="-1"
          >
            <span
              aria-label="ellipsis"
              class="anticon anticon-ellipsis"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="ellipsis"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
                />
              </svg>
            </span>
            <i
              class="ant-menu-submenu-arrow"
            />
          </div>
        </li>
      </ul>
      <div
        aria-hidden="true"
        style="display: none;"
      />
    </div>
  </body>,
  "container": <div>
    <ul
      class="ant-menu-overflow ant-menu ant-menu-root ant-menu-horizontal ant-menu-light bg-eeric-black-light w-fit font-primary text-xl css-dev-only-do-not-override-1adbn6x"
      data-menu-list="true"
      role="menu"
      tabindex="0"
    >
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-selected ant-menu-item-only-child text-white"
        data-menu-id="rc-menu-uuid-test-home"
        role="menuitem"
        style="opacity: 1; order: 0;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/"
          >
            Home
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child text-white"
        role="menuitem"
        style="opacity: 1; order: 1;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/sports"
          >
            Sports
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child text-white"
        role="menuitem"
        style="opacity: 1; order: 2;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/medal"
          >
            Medal
          </a>
        </span>
      </li>
      <li
        class="ant-menu-overflow-item ant-menu-item ant-menu-item-only-child text-white"
        role="menuitem"
        style="opacity: 1; order: 3;"
        tabindex="-1"
      >
        <span
          class="ant-menu-title-content"
        >
          <a
            href="/audience"
          >
            Audience
          </a>
        </span>
      </li>
      <li
        aria-hidden="true"
        class="ant-menu-overflow-item ant-menu-overflow-item-rest ant-menu-submenu ant-menu-submenu-horizontal"
        role="none"
        style="opacity: 0; height: 0px; overflow-y: hidden; order: 9007199254740991; pointer-events: none; position: absolute;"
      >
        <div
          aria-controls="rc-menu-uuid-test-rc-menu-more-popup"
          aria-expanded="false"
          aria-haspopup="true"
          class="ant-menu-submenu-title"
          data-menu-id="rc-menu-uuid-test-rc-menu-more"
          role="menuitem"
          tabindex="-1"
        >
          <span
            aria-label="ellipsis"
            class="anticon anticon-ellipsis"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="ellipsis"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M176 511a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0zm280 0a56 56 0 10112 0 56 56 0 10-112 0z"
              />
            </svg>
          </span>
          <i
            class="ant-menu-submenu-arrow"
          />
        </div>
      </li>
    </ul>
    <div
      aria-hidden="true"
      style="display: none;"
    />
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
